### PR TITLE
Add FXIOS-9459 [Microsurvey] Add initial UI tests

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9459)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20943)

## :bulb: Description
Add initial UI tests for microsurvey tests to assist automation team on writing UI tests for microsurveys. The microsurvey feature requires a trigger and a reset to show the survey and I added `generateTriggerForMicrosurvey` to set this up for the tests.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by isabelrios** [comment]
@cyndichin thanks for this PR adding new UI Tests 🎉 

I would take a deeper look, but at a first glance, we would need to specify the test suite where we want to run these tests. 
imagen

We have the smoketest that runs for every PR, and the full functional that runs daily. I would recommend to add the tests to the latter, and then after checking with SV we can change and add some test(s) (the most relevant) to one of the smoketest.

The tests are running in the four smoketests now. So you would need to change that.

Go to each test plan, except for the full functional, and disable the test suite:
imagen

Let us know if you have doubts or questions! cc @clarmso @dragosb01

**Review by isabelrios — CHANGES_REQUESTED** [review]
As per comment about where the tests are running.

**Comment by cyndichin** [comment]
@isabelrios thank you for the callout! I have removed the microsurvey tests from all suites expect the `FullFunctionalTestsPlan`. Let me know if there are any other issues!

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/MicrosurveyTests.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+final class MicrosurveyTests: BaseTestCase {
+    override func setUp() {
+        launchArguments = [
+            LaunchArguments.SkipIntro,
+            LaunchArguments.ResetMicrosurveyExpirationCount
+        ]
+        super.setUp()
+    }
+
+    func testShowMicrosurveyPromptFromHomepageTrigger() {
```
Comment (dragosb01) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/MicrosurveyTests.swift:16]: I am wondering if these tests are matching scenarios with testrail test cases. If not we could add the missing links in the future.

Comment (isabelrios) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/MicrosurveyTests.swift:16]: yeah, that's why I commented in our slack channel. Currently draft version of tests are [here](https://testrail.stage.mozaws.net/index.php?/suites/view/53811&group_by=cases:section_id&group_order=asc&display_deleted_cases=0) they will be moved to the FullFunctional test suite when ready, and then we can match them to add the link.
If you can take a look and see if we should modify something while being in draft mode, that would help when they are ready. Thanks!

**Comment by isabelrios** [comment]
I saw a few tries to rerun the build due to unit test failures, @cyndichin remember to do that from bitrise UI pipeline view, otherwise you run one workflow independently that does not have all the dependencies and so it fails

**Comment by cyndichin** [comment]
> I saw a few tries to rerun the build due to unit test failures, @cyndichin remember to do that from bitrise UI pipeline view, otherwise you run one workflow independently that does not have all the dependencies and so it [fails](https://app.bitrise.io/build/d958cbdc-6011-4e5c-a171-60e31f9b6814)

ah thanks! I must have pressed the wrong button, thanks for the callout! Just reran the entire pipeline.

**Review by dragosb01 — APPROVED** [review]
To me it looks good. We could make the proper adjustment in the future if needed.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/MicrosurveyTests.swift (hunk 1 above)
Comment (isabelrios) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/MicrosurveyTests.swift:16]: We will match them, no worries about that

**Review by isabelrios — APPROVED** [review]

**Pull request merged**